### PR TITLE
main: delay reporing "Language already defined" fatal error till parsing starts

### LINKED ISSUE #2935: main: ctags option processing fails if "Language already defined", including ctags --help
Hello,

recently gentoo updated to ctags-20210321.0. Since this version added the BibTeX parser but I had the file ~/.ctags.d/BibTeX.ctags I got the message 'ctags: Language "BibTeX" already defined'. So far so good - but 'ctags --help' or 'ctags --list-languages' only outputs the message above and apparently bails out of option processing. I think that at least such info requests should continue to work.

```
$ ctags --version
ctags: Language "BibTeX" already defined

After removing ~/.ctags.d/BibTeX.ctags:
$ ctags --version
Universal Ctags 5.9.0, Copyright (C) 2015 Universal Ctags Team
Universal Ctags is derived from Exuberant Ctags.
Exuberant Ctags 5.8, Copyright (C) 1996-2009 Darren Hiebert
Compiled: Mar 30 2021, 09:48:49
URL: https://ctags.io/
Optional compiled features: +wildcards, +regex, +iconv, +option-directory, +xpath, +sandbox, +packcc
```

How do you get ctags binary:

via gentoo portage

### PR DESCRIPTION
**Pull request by masatake**
Close #2935 again.

The original version of this pull request was merged and reverted because a segmentation fault is detected on AppVeyor.

Newly introduced vStringNewPrintf and vStringNewVPrintf trigger a bug in vStringNCatS.
This pull request includes a fix for the bug.

### CODE REVIEW

**Comment by masatake** [comment]
The segmentation faults are still reproduced on 32bit platforms.

**Comment by masatake** [comment]
The bug is related to 646e26b7468970e18293d6cd0306a5482850a1ff.

File: main/vstring.c [hunk 1]
```diff
@@ -124,9 +124,17 @@ extern void vStringNCat (
 extern void vStringNCatS (
 		vString *const string, const char *const s, const size_t length)
 {
-	size_t len = strlen (s);
+	size_t len = length;
```
Comment (masatake) [inline main/vstring.c:159]: strnlen is available on Pois.

**Comment by leleliu008** [comment]
after adding `#include <stdarg.h>` to `main/vstring.h`, the NetBSD and OpenBSD GitHub Actions workflow error will be missing.

It seems that the macOS GitHub Actions workflow error is caused by GitHub Actions Team, not a issue from HomeBrew. [Reference](https://github.com/Homebrew/homebrew-core/issues/135315), I think it will be fixed later.

**Comment by masatake** [comment]
@leleliu008 thank you.

The failure on "Environment: compiler=msvc, ARCH=x64" on Appveyor implies something critical bug. I should inspect the failure more. However, for making a new release, the change I illustrated in https://github.com/universal-ctags/ctags/issues/2935#issuecomment-1568998935 may be enough.

**Comment by masatake** [comment]
Surprisingly https://github.com/universal-ctags/ctags/issues/2935#issuecomment-1568998935 doesn't work at all. I must find another solution.

**Comment by masatake** [comment]
I'll make a pull request based on https://github.com/universal-ctags/ctags/issues/2935#issuecomment-1568998935  for closing #2935.

**Pull request closed without merging**